iosapp/lib-ios/Sources/LibFilter/FilterProxy.swift

```diff
@@ -2,66 +2,77 @@ import Dependencies
 import GertieIOS
 import LibClients
 import LibCore
+import XCore
 
 #if canImport(NetworkExtension)
   import NetworkExtension
 #else
   public struct NEProviderStopReason {} // CI
 #endif
 
-public class FilterProxy {
-  @Dependency(\.osLog) var logger
-  @Dependency(\.storage) var storage
-  @Dependency(\.suspendingClock) var clock
-  @Dependency(\.date.now) var now
-  @Dependency(\.calendar) var calendar
+public final class FilterProxy: Sendable {
+  struct Deps: Sendable {
+    @Dependency(\.osLog) var logger
+    @Dependency(\.storage) var storage
+    @Dependency(\.suspendingClock) var clock
+    @Dependency(\.date.now) var now
+    @Dependency(\.calendar) var calendar
+  }
+
+  struct HeartbeatDurations: Sendable {
+    var normal: Duration
+    var current: Duration
+  }
 
-  var protectionMode: ProtectionMode
-  var heartbeatTask: Task<Void, Error>?
-  var normalHeartbeatInterval: Duration
-  var hearbeatInterval: Duration = .seconds(10)
+  let deps = Deps()
+  let protectionMode: Mutex<ProtectionMode>
+  let heartbeatDurations: Mutex<HeartbeatDurations>
+  let heartbeatTask: Mutex<Task<Void, Error>?>
```

Comment:
this is where i'm using the backport i introduced of the apple (iOS18+) `Mutex` type. eventually i think i want to replace my old `Core.Mutex` used in the Mac app, but that can wait another day.